src/language-js/embed/utils.js

```diff
@@ -66,8 +66,18 @@ function hasLanguageComment(node, languageName) {
   );
 }
 
+function isAsConstExpression(node) {
+  return (
+    node.type === "TSAsExpression" &&
+    node.typeAnnotation.type === "TSTypeReference" &&
+    node.typeAnnotation.typeName.type === "Identifier" &&
+    node.typeAnnotation.typeName.name === "const"
+  );
```

Comment:
We already merge  https://github.com/prettier/prettier/pull/15130?notification_referrer_id=NT_kwDOAAKiKLE3MTA1ODk0MDM3OjE3MjU4NA#event-10389275756, so we probably should add support for Flow parser too.